app/service/file_svc.py

```diff
@@ -16,6 +16,11 @@ def __init__(self, plugins, exfil_dir):
         self.plugins = plugins
         self.exfil_dir = exfil_dir
         self.log = self.add_service('file_svc', self)
+        self.data_svc = self.get_service('data_svc')
```

Comment:
So close to perfect! Instead of creating a new init function, move line 23 into the already existing __init__ function.

Reply:
Do you mean the "__init__" function of file_svc?
I checked some discussions about setting class attribute with await in __init__, to implement it is adding "async def init" function for it.
or did I miss something?

Reply:
ah- you're 100% right. In that case, let's do this:

create a new private function called "_verify_file", which should be called as the first line of the "download(..)" function. in that function, we can check if the name is not empty AND if the platform is valid. Instead of having the valid_platforms in your init function, let's put it in the _verify_file() function. even though we'll recalculate on each downloaded file, a) I think it was more readable and b) it'll be more flexible for some changes we're planning to allow new abilities to be added without restarting caldera. sorry for misdirecting you on my last comments about this-- you had it that way in your previous commit and I had you change it!